protocol/src/main/java/org/jboss/as/protocol/mgmt/ManagementClientChannelStrategy.java

```diff
@@ -88,13 +88,53 @@ public static ManagementClientChannelStrategy create(final ProtocolConnectionCon
     /**
      * Create a new establishing management client channel-strategy
      *
+     * @param baseConfig the base connection configuration
+     * @param handler the {@code ManagementMessageHandler}
+     * @param cbHandler a callback handler
+     * @param saslOptions the sasl options
+     * @param sslContext the ssl context
+     * @param closeHandler a close handler
+     * @param blockOnClose {@code true} if {@linkplain ManagementClientChannelStrategy#close() close} should block
+     *                                  until the connection manager is shutdown, otherwise the shutdown is done
+     *                                  asynchronously
+     * @return the management client channel strategy
+     */
+    public static ManagementClientChannelStrategy create(final ProtocolConnectionConfiguration baseConfig,
+                                                         final ManagementMessageHandler handler,
+                                                         final CallbackHandler cbHandler,
+                                                         final Map<String, String> saslOptions,
+                                                         final SSLContext sslContext,
+                                                         final CloseHandler<Channel> closeHandler,
+                                                         final boolean blockOnClose) {
+        return create(createConfiguration(baseConfig, saslOptions, cbHandler, sslContext), ManagementChannelReceiver.createDelegating(handler), closeHandler, blockOnClose);
```

Comment:
Even though it is public this is an internal class so I'd rather just add a param to the existing methods than create new ones.

This management comms code is like a fat man struggling to lose weight, so no twinkies!